test/peer_tests.ml

```diff
@@ -0,0 +1,38 @@
+open Lwt.Infix
+open Commons
+
+module Peer_tests = struct
+  open Pollinate
+  let add_peer_test () =
+    let open Commons in
+    let peer_b = Peer.add_peer peer_a peer_b in
```

Comment:
I think this is not correct. Actual `add_peer` function return a new peer, should it work this way or should I use `ref`?

Reply:
Totally reworked it.